File: src/services/preview.ts
/** 文件预览的命令封装。对应后端 `commands::preview`。 */

import type { PreviewPayload } from "../types/preview";
import { invokeCmd } from "./tauri";

/** 请求指定路径的预览信息；后端按扩展名路由到 text/image/zip 处理。 */
export function requestPreview(filePath: string) {
  return invokeCmd<PreviewPayload>("request_preview", { filePath });
}

File: src-tauri/src/commands/preview.rs
//! 文件预览命令。

use std::sync::Arc;

use tauri::State;

use crate::{app_state::AppState, services::preview};

/// 前端请求预览指定路径；后端按扩展名路由到 text/image/zip 解析，
/// 不支持的类型返回 `{"type":"unsupported"}`。
#[tauri::command]
pub fn request_preview(
    state: State<'_, Arc<AppState>>,
    file_path: String,
) -> Result<serde_json::Value, String> {
    preview::detect_preview(&state.db_path, &file_path).map_err(|e| e.to_string())
}
